File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        # if l1 == None:
        #     return l2
        # if l2 == None:
        #     return l1
        # if l2.val >= l1.val:
        #     l1.next = self.mergeTwoLists(l1.next,l2)
        #     return l1
        # else:
        #     l2.next = self.mergeTwoLists(l2.next,l1)
        #     return l2
        final = ListNode(-1)
        p = final
        while l1 != None and l2 != None:
            if l2.val >= l1.val:
                p.next = l1
                l1 = l1.next
            else:
                p.next = l2
                l2 = l2.next
            p = p.next
        if l1 == None:
            p.next = l2
        else:
            p.next = l1
        return final.next

File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode p1, ListNode p2) {
        
        ListNode res = new ListNode() ;
        ListNode ans = res;
        while (p1 != null && p2 != null){
            if (p1.val < p2.val){
                ans.next = p1;
                p1 = p1.next;
                ans = ans.next;
            } else {
                ans.next = p2;
                p2 = p2.next ; 
                ans = ans.next;
            }
        }
        if (p1 == null)
            ans.next = p2;
        if (p2 == null)
            ans.next = p1;
        return res.next;
    }
}